src/modules/python/pbs_hooks/PBS_cray_jacs.PY

```diff
@@ -273,6 +273,49 @@ def merge_dict(base, new):
     return merged
 
 
+def retry_post(data):
+    """
+    In the case where a POST fails due to a 400 error,
+    it could be because there is already a job on the cray side.
+    In that case, we should try to delete the existing job and
+    resubmit a new one.
+    """
+    event = pbs.event()
+    jid = event.job.id
+
+    joburl = HookHelper.build_path(resource='job', jobid=jid)
+    del_timeout = HookHelper.get_config()['delete_timeout']
+    try:
+        r2 = delete(joburl, timeout=del_timeout)
+        r2.raise_for_status()
+    except requests.Timeout:
+        log_with_caller(pbs.EVENT_ERROR, 'DELETE timed out')
+        raise OfflineError('Job delete timed out')
+    except requests.HTTPError:
+        # If 404, then maybe the job that was there is now gone,
+        # try posting again. Otherwise, raise an OfflineError
+        if r2.status_code != 404:
+            log_with_caller(pbs.EVENT_ERROR, 'DELETE job failed')
+            raise OfflineError('Job delete failed')
+
+    url = HookHelper.build_path(resource='job')
+    post_timeout = HookHelper.get_config()['post_timeout']
+    try:
+        r3 = post(url, json=data, timeout=post_timeout)
+        r3.raise_for_status()
```

Comment:
You might want to rename these from r2 and r3 to r1 and r2 or rdelete and rpost or something.  Starting something at 2 is weird.